### ISSUE TITLE
Timex.Timezone.convert false behaviour since 3.6 with CEST when DST changes from sommer to winter

### ISSUE THREAD

**Issue by lumixmtb** [bitwalker/timex#601]
### Steps to reproduce

```
Timex.parse!("2020-10-25T00:00:00Z", "{ISO:Extended}")  
|> Timex.Timezone.convert("Europe/Berlin")
```

### Description of issue

With 3.4.1:
```
iex(7)> ref = Timex.parse!("2020-10-25T00:00:00Z", "{ISO:Extended}")     
~U[2020-10-25 00:00:00Z]
iex(8)> ref |> Timex.Timezone.convert("Europe/Berlin")
#DateTime<2020-10-25 02:00:00+02:00 CEST Europe/Berlin>
```
with 3.6.1:
```
9:26 AM] iex(17)> ref = Timex.parse!("2020-10-25T00:00:00Z", "{ISO:Extended}")
~U[2020-10-25 00:00:00Z]
iex(18)> ref |> Timex.Timezone.convert("Europe/Berlin")              
#DateTime<2020-10-25 01:00:00+01:00 CET Europe/Berlin>
```

This new behaviour in 3.6.1 is false. The timestamp 01:00+01 CET does not exist (should be 02:00+02 CEST). The DST changes at 3:00 local time where the time is shifted back an hour to 02:00

This should be the expected behaviour when DST switches from summer to winter:
UTC        Local UTC+01
22         00+02 CEST
23         01+02 CEST
00         02+02 CEST
01         02+01 CET
02         03+01 CET
03         04+01 CET

Does someone knows what could have cause this change?
Thanks
(not tested with 3.5)

**Comment by jarrodmoldrich** [comment]
I'm not sure if this is related, but I found this discrepancy between the results of `Timex.Timezone.convert/2` `DateTime.shift_zone!/3` when converting a UTC value at the onset of daylight savings time.

Here are the time periods for 2020 for `America/Santiago`:
```
[
  {63753274800,
   %{
     from: %{standard: 63753260400, utc: 63753274800, wall: 63753260400},
     std_off: 0,
     until: %{standard: 63766569600, utc: 63766584000, wall: 63766569600},
     utc_off: -14400,
     zone_abbr: "-04"
   }},
  {63766584000,
   %{
     from: %{standard: 63766569600, utc: 63766584000, wall: 63766573200},
     std_off: 3600,
     until: %{standard: 63784710000, utc: 63784724400, wall: 63784713600},
     utc_off: -14400,
     zone_abbr: "-03"
   }}
]
```

and here are the related queries for `shift_zone` where the time change happens at the `utc` provided by `Tzdata`:

```
iex(101)> Timex.from_unix(63753274800-62167219200-1) |> DateTime.shift_zone!("America/Santiago")
#DateTime<2020-04-04 23:59:59-03:00 -03 America/Santiago>
iex(102)> Timex.from_unix(63753274800-62167219200) |> DateTime.shift_zone!("America/Santiago")
#DateTime<2020-04-04 23:00:00-04:00 -04 America/Santiago>
```

and here are the ones for `convert` which appears to kick in an hour late:

```
iex(103)> Timex.from_unix(63753274800-62167219200-1) |> Timex.Timezone.convert("America/Santiago")
#DateTime<2020-04-04 23:59:59-03:00 -03 America/Santiago>
iex(104)> Timex.from_unix(63753274800-62167219200) |> Timex.Timezone.convert("America/Santiago")
#DateTime<2020-04-05 00:00:00-03:00 -03 America/Santiago>
iex(105)> Timex.from_unix(63753274800-62167219200+3599) |> Timex.Timezone.convert("America/Santiago")
#DateTime<2020-04-05 00:59:59-03:00 -03 America/Santiago>
iex(106)> Timex.from_unix(63753274800-62167219200+3600) |> Timex.Timezone.convert("America/Santiago")
#DateTime<2020-04-05 00:00:00-04:00 -04 America/Santiago>
```

Hope this helps to track this down in some way.

**Comment by bitwalker** [comment]
This has been fixed as of 3.7
